nextcloud/android
# Feature Compatible File Names

**How to Test?**

For all tests should do checks according to OC Capability. e.g. If forbiddenFilenames is false or unknown, no need to check reserved Windows names.

1. [**Create a New Document]** Tap the plus button on the home screen, select "New Document," and enter a file name for the new document.
2. [**Rename a File]** Tap the three dots icon at the end of any file in the list, and select "Rename."
3. [**Rename a File From File Details]** Go to "File Details". From the top right, tap the three dots icon, select "Rename," and test it.
4. [**Create a New Folder]** Tap the plus button on the home screen, select "New Folder," and enter a name for the new folder.
5. [**Move or Copy file to Any Folder]** Tap the three dots icon at the end of any file in the list, and select "Move" or "Copy." Attempt to move or copy to both an invalid and a valid folder path.
6. [**Share a File From Other Any App]** From any other app, try to share a file with the Nextcloud Files app. If the shared file contains invalid characters or reserved name, user must receive feedback, and the operation should not be allowed.
7. [**Upload Contents From Other Apps]** Tap the plus button on the home screen, select "Upload Contents from Other Apps." If the contents include a path, the user must receive feedback, and the operation should not be allowed.
8. [**Create a New File From Other File Manager Apps]** User can create forbidden file names from native files app such as Google Files.
9. **[Create a New Template]** User may have a template for creating office files such as new presentation and new text document etc.
